tiles/src/main/java/com/protomaps/basemap/layers/Buildings.java

```diff
@@ -93,13 +98,46 @@ public void processOsm(SourceFeature sf, FeatureCollector features) {
       // Names should mostly just be for POIs
       // Sometimes building name and address are useful items, but only at zoom 17+
       //OsmNames.setOsmNames(feature, sf, 13);
+    } else if (sf.hasTag("addr:housenumber")) {
+      FeatureCollector.Feature feature = null;
+      if (sf.isPoint()) {
+        feature = features.point(this.name());
+      } else if (sf.canBePolygon()) {
+        feature = features.centroid(this.name());
+      }
+      if (feature != null) {
+        feature
+          .setId(FeatureId.create(sf))
+          .setAttr("addr_housenumber", sf.getString("addr:housenumber"))
+          .setAttr("addr_street", sf.getString("addr:street"))
+          .setAttr("kind", "address")
+          .setMinZoom(15);
+      }
     }
   }
 
   @Override
   public List<VectorTile.Feature> postProcess(int zoom, List<VectorTile.Feature> items) throws GeometryException {
     if (zoom == 15) {
-      return items;
+      List<VectorTile.Feature> buildings = new ArrayList<>();
+
+      // deduplicate addresses
+      HashMap<Map<String, Object>, List<VectorTile.Feature>> groupedAddresses = new LinkedHashMap<>();
```

Comment:
Is this meant to address one building that is labeled twice on the map? Is there a specific place in the world where that problem is pervasive or is it an edge case where maybe it's a data problem in OSM?

Reply:
NVM, I see the issue is that it also includes POIs that have an `addr:housenumber` in OSM. I think for the first pass we should limit the house number displays to just those tagged on buildings and not POIs, because de-duplicating them in post-process is going to lead to non-deterministic results potentially, and ideally POIs will be labeled by an icon symbology instead of just a housenumber. What do you think? Labeling on OSM Carto seems to be focused on the building `addr:housenumber` too.